src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -116,13 +125,33 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                 }
             }
         }
+        if (itemsPerTopic.Count == 0)
+            return count;
+        QueueWithItems<KafkaItem> currentTopic = itemsPerTopic[0];
```

Comment:
This can be moved below and the first assignment should be inside `foreach` loop so `null` assignment in initialization

Reply:
done, moved it down